lib/Reference/OfficeTargetReferenceProvider.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace OCA\Richdocuments\Reference;
+
+use Exception;
+use OCA\Richdocuments\AppInfo\Application;
+use OCA\Richdocuments\Service\FileTargetService;
+use OCP\Collaboration\Reference\ADiscoverableReferenceProvider;
+use OCP\Collaboration\Reference\IReference;
+use OCP\Collaboration\Reference\Reference;
+use OCP\Files\IMimeTypeDetector;
+use OCP\Files\IRootFolder;
+use OCP\IL10N;
+use OCP\IPreview;
+use OCP\IURLGenerator;
+use Psr\Log\LoggerInterface;
+
+class OfficeTargetReferenceProvider extends ADiscoverableReferenceProvider {
+
+	public function __construct(
+		private FileTargetService $fileTargetService,
```

Comment:
Constructor property promotion seems to be only allowed starting with PHP 8.0. But since php 8.0 is set as platform requirement in `composer.json` and this feature targets only Nextcloud 27 and later, this seems fine.

Reply:
Yes, only targeting 27+ with that.